cmd/cli/serve/util.go

```diff
@@ -143,84 +177,43 @@ func GetRequesterConfig(ctx context.Context, cfg types.BacalhauConfig, createJob
-		ClusterPort:              cfg.Cluster.Port,
-		ClusterAdvertisedAddress: cfg.Cluster.AdvertisedAddress,
-		ClusterPeers:             cfg.Cluster.Peers,
-	}, nil
-}
-
-func getExecutionStore(ctx context.Context, storeCfg types.JobStoreConfig, path string) (store.ExecutionStore, error) {
-	if err := storeCfg.Validate(); err != nil {
-		return nil, err
-	}
-
-	switch storeCfg.Type {
-	case types.BoltDB:
-		return boltdb.NewStore(ctx, path)
-	default:
-		return nil, fmt.Errorf("unknown JobStore type: %s", storeCfg.Type)
-	}
-}
-
-func getJobStore(ctx context.Context, storeCfg types.JobStoreConfig, path string) (jobstore.Store, error) {
-	if err := storeCfg.Validate(); err != nil {
-		return nil, err
+func getNetworkConfig(cfg types2.Bacalhau) (node.NetworkConfig, error) {
+	// TODO(review) see method getPublicNATSOrchestratorURL for why we ignore the host.
+	_, listenPortStr, err := net.SplitHostPort(cfg.Orchestrator.Listen)
+	if err != nil {
+		return node.NetworkConfig{}, err
 	}
-
-	switch storeCfg.Type {
-	case types.BoltDB:
-		log.Ctx(ctx).Debug().Str("Path", path).Msg("creating boltdb backed jobstore")
-		return boltjobstore.NewBoltJobStore(path)
-	default:
-		return nil, fmt.Errorf("unknown JobStore type: %s", storeCfg.Type)
+	listenPort, err := strconv.ParseInt(listenPortStr, 10, 64)
+	if err != nil {
+		return node.NetworkConfig{}, err
 	}
-}
-
-func getNodeID(ctx context.Context, nodeNameProviderType string) (string, error) {
-	nodeNameProviders := map[string]idgen.NodeNameProvider{
-		"hostname": idgen.HostnameProvider{},
-		"aws":      idgen.NewAWSNodeNameProvider(),
-		"gcp":      idgen.NewGCPNodeNameProvider(),
-		"uuid":     idgen.UUIDNodeNameProvider{},
-		"puuid":    idgen.PUUIDNodeNameProvider{},
+	storeDir, err := cfg.NetworkTransportDir()
+	if err != nil {
+		return node.NetworkConfig{}, err
 	}
-	nodeNameProvider, ok := nodeNameProviders[nodeNameProviderType]
-	if !ok {
-		return "", fmt.Errorf(
-			"unknown node name provider: %s. Supported providers are: %s", nodeNameProviderType, lo.Keys(nodeNameProviders))
+	_, clusterListenPortStr, err := net.SplitHostPort(cfg.Orchestrator.Cluster.Listen)
+	if err != nil {
+		return node.NetworkConfig{}, err
 	}
-
-	nodeName, err := nodeNameProvider.GenerateNodeName(ctx)
+	clusterListenPort, err := strconv.ParseInt(clusterListenPortStr, 10, 64)
 	if err != nil {
-		return "", err
+		return node.NetworkConfig{}, err
 	}
-
-	return nodeName, nil
+	return node.NetworkConfig{
+		Port:              int(listenPort),
+		AdvertisedAddress: cfg.Orchestrator.Advertise,
+		Orchestrators:     cfg.Compute.Orchestrators,
+		StoreDir:          storeDir,
+		// TODO(review): an auth secret is no longer part of the config, should we operate without one, or include it
+		// in the config?
+		AuthSecret: "TODO",
+		// TODO(review): a cluster name is no longer part of the config, should we operate without one, or include it
+		// in the config?
+		ClusterName:              "TODO",
```

Comment:
@wdbaruni The new configuration no longer has fields for `ClusterName` and `AuthSecret`, is the intention to leave these blank and permit configuration of them later as part of https://github.com/bacalhau-project/bacalhau/issues/3850?